packages/ui/SingleResponseCard/actions.ts

```diff
@@ -24,26 +24,35 @@ export const createTagAction = async (environmentId: string, tagName: string) =>
   const isAuthorized = await hasUserEnvironmentAccess(session.user!.id, environmentId);
   if (!isAuthorized) throw new AuthorizationError("Not authorized");
 
+  const { hasCreateOrUpdateAccess } = await verifyUserRoleAccess(environmentId, session.user!.id);
+  if (!hasCreateOrUpdateAccess) throw new AuthorizationError("Not authorized");
+
   return await createTag(environmentId, tagName);
 };
 
-export const createTagToResponeAction = async (responseId: string, tagId: string) => {
+export const createTagToResponeAction = async (responseId: string, tagId: string, environmentId: string) => {
   const session = await getServerSession(authOptions);
   if (!session) throw new AuthorizationError("Not authorized");
 
   const isAuthorized = await canUserAccessTagOnResponse(session.user!.id, tagId, responseId);
   if (!isAuthorized) throw new AuthorizationError("Not authorized");
 
+  const { hasCreateOrUpdateAccess } = await verifyUserRoleAccess(environmentId, session.user!.id);
```

Comment:
we have the same action problem as mentioned below. Please derive environmentId instead of param.